File: January 2025/13.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
  public:
    int maxWater(vector<int> &arr) {
        int left = 0, right = arr.size() - 1;
        int max_water = 0;

        while (left < right) {
            int width = right - left;
            int height = min(arr[left], arr[right]);
            int current_water = width * height;
            max_water = max(max_water, current_water);

            if (arr[left] < arr[right]) {
                left++;
            } else {
                right--;
            }
        }

        return max_water;
    }
};

File: January 2025/13.py
class Solution:
    def maxWater(self, arr):
        left, right = 0, len(arr) - 1
        max_water = 0
        
        while left < right:
            width = right - left
            height = min(arr[left], arr[right])
            current_water = width * height
            max_water = max(max_water, current_water)
            
            if arr[left] < arr[right]:
                left += 1
            else:
                right -= 1
        
        return max_water
